engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/UserTaskTest.java

```diff
@@ -418,6 +420,191 @@ public void shouldCreateJobWithEvaluatedCandidateUsersExpressionHeader() {
+            .create();
+
+    // then
+    final Record<JobRecordValue> job =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    final Map<String, String> customHeaders = job.getValue().getCustomHeaders();
+    assertThat(customHeaders).hasSize(0).doesNotContainKey(Protocol.USER_TASK_DUE_DATE_HEADER_NAME);
+  }
+
+  @Test
+  public void shouldCreateJobWithFollowUpDateHeader() {
+    // given
+    ENGINE.deployment().withXmlResource(process(t -> t.zeebeFollowUpDate(FOLLOW_UP_DATE))).deploy();
+
+    // when
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // then
+    final Record<JobRecordValue> job =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    final Map<String, String> customHeaders = job.getValue().getCustomHeaders();
+    assertThat(customHeaders)
+        .hasSize(1)
+        .containsEntry(Protocol.USER_TASK_FOLLOW_UP_DATE_HEADER_NAME, FOLLOW_UP_DATE);
+  }
+
+  @Test
+  public void shouldCreateJobWithEvaluatedFollowUpDateExpressionHeader() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(process(t -> t.zeebeFollowUpDateExpression("followUpDate")))
+        .deploy();
+
+    // when
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withVariables(Map.of("followUpDate", FOLLOW_UP_DATE))
+            .create();
+
+    // then
+    final Record<JobRecordValue> job =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    final Map<String, String> customHeaders = job.getValue().getCustomHeaders();
+    assertThat(customHeaders)
+        .hasSize(1)
+        .containsEntry(Protocol.USER_TASK_FOLLOW_UP_DATE_HEADER_NAME, FOLLOW_UP_DATE);
+  }
+
+  @Test
+  public void shouldCreateJobAndIgnoreEmptyFollowUpDateHeader() {
+    // given
+    ENGINE.deployment().withXmlResource(process(t -> t.zeebeFollowUpDate(""))).deploy();
+
+    // when
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // then
+    final Record<JobRecordValue> job =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    final Map<String, String> customHeaders = job.getValue().getCustomHeaders();
+    assertThat(customHeaders)
+        .hasSize(0)
+        .doesNotContainKey(Protocol.USER_TASK_FOLLOW_UP_DATE_HEADER_NAME);
+  }
+
+  @Test
+  public void shouldCreateJobAndIgnoreEmptyEvaluatedFollowUpDateExpressionHeader() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(process(t -> t.zeebeDueDateExpression("=followUpDate")))
```

Comment:
:x: Looks like this should be `zeebeFollowUpDateExpression`.

Reply:
Whoops! Nice catch!